consensus/polybft/consensus_runtime.go

```diff
@@ -856,6 +856,29 @@ func (c *consensusRuntime) GenerateExitProof(exitID, epoch, checkpointBlock uint
 	return tree.GenerateProofForLeaf(e, 0)
 }
 
+// GetStateSyncProof returns the proof of the bundle for the state sync
+func (c *consensusRuntime) GetStateSyncProof(stateSyncID uint64) (*types.StateSyncProof, error) {
+	bundlesToExecute, err := c.state.getBundles(stateSyncID, 1)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get bundles: %w", err)
+	}
+
+	if len(bundlesToExecute) == 0 {
+		return nil, fmt.Errorf("cannot find bundle containing StateSync id %d", stateSyncID)
+	}
+
+	for _, bundle := range bundlesToExecute[0].StateSyncs {
```

Comment:
Could we remove that `BundleProof` struct we had earlier, and when we build bundles, just directly build this new `StateSyncProof`, or will we do it in another PR?
It would even be easier to save these `StateSyncProof`s in `boltdb` and get them when `GetStateSyncProof` is called, because its key would be the `StateSync.ID`.

Reply:
We are going to do that, in RPC we're only exposing StateSyncs, I can start right away with the bigger refactor after merging the complete workflow we have for the demo.